include/z64actor.h

```diff
@@ -96,7 +96,7 @@ typedef struct {
     /* 0x0C */ ActorShadowFunc shadowDraw; // Shadow draw function
     /* 0x10 */ f32 shadowScale; // Changes the size of the shadow
     /* 0x14 */ u8 shadowAlpha; // Default is 255
-    /* 0x15 */ u8 feetFloorFlags; // Set if the actor's foot is clipped under the floor. & 1 is right foot, & 2 is left
+    /* 0x15 */ u8 feetFloorFlag; // 0 if actor or feet aren't on ground, or 1 or 2 depending on feet positions
```

Comment:
Since this technically stores 2 flags (& 1 and & 2), I feel like having it plural makes more sense

Reply:
It's actually just a value, it's always 0, 1 or 2 (except temporarily where the feet positions are computed), never 1 | 2

Reply:
Also I can't make sense of the conditional that determines if the value should be 1 or 2 in the case where both feet touch the ground, and since the only use in the game is a single != 0 check in player it's hard to tell what they were going for

Reply:
I see the game expects it to be 3 at some point, so both flags are definitely intended to be set at the same time (at least temporarily) and it would seem wrong to treat it as a single value, but I'm not too familiar with this stuff so idk

Reply:
I'm not seeing this value as a bitfield but as

- 0 if actor or feet aren't on ground
- 1 if left feet is "more something" than right feet
- 2 if right feet is "more something" than left feet

the "more something" being unclear, especially due to the case where  `feetFloorFlag = 3` temporarily. I don't understand the conditional that tells apart 1 and 2:
https://github.com/Dragorn421/oot/blob/2e45e9ab8249df74cbe5a24287f4597fa81bad39/src/code/z_actor.c#L195-L203
```c
        } else if (actor->shape.feetFloorFlag == 3) {
            f32 footDistY = actor->shape.feetPos[FOOT_LEFT].y - actor->shape.feetPos[FOOT_RIGHT].y;

            if ((floorHeight[FOOT_LEFT] + footDistY) < (floorHeight[FOOT_RIGHT] - footDistY)) {
                actor->shape.feetFloorFlag = 2;
            } else {
                actor->shape.feetFloorFlag = 1;
            }
        }
```

Reply:
Took a look at it.

Prior to that code a `for` loop that initializes actor->shape.feetFloorFlag. It sets the value to 0 if `(-1 <= distToFloor <= 0)` is false for both feet, 2 if only true for the left foot, 1 if only true for the right foot, and 3 if true for both feet. Thus, 3 means that both feet are planted on the ground.

Not fully sure what the `actor->shape.feetFloorFlag == 3` case is attempting to accomplish, but it seems to do this:
`footDistY` stores the diff between the left and right foot's shape.y position. if the left foot shape is lower than the right foot, then the condition effectively becomes

```python
# Left foot lower than right
if (left floor - abs(footDiff)) < (right floor + abs(footDiff))
    feetFloorFlag = LEFT_FOOT // 2
else
    feetFloorFlag = RIGHT_FOOT // 1
````

else if the right foot shape is lower than the left foot, the condition becomes

```python
# right foot lower than left
if (left floor + abs(footDiff)) < (right floor - abs(footDiff))
    feetFloorFlag = LEFT_FOOT // 2
else
    feetFloorFlag = RIGHT_FOOT // 1
````